designer/report-designer-extension-pentaho/src/main/java/org/pentaho/reporting/designer/extensions/pentaho/repository/util/PublishUtil.java

```diff
@@ -144,17 +140,18 @@ public static void launchReportOnServer( final String baseUrl, final String path
     }
   }
 
-  public static int publish( final byte[] data, final String path, final AuthenticationData loginData )
-    throws IOException {
+  public static int publish( final byte[] data, final String path, final AuthenticationData loginData, final Optional<Properties> fileProperties ) throws IOException {
     int responseCode = HTTP_RESPONSE_FAIL;
```

Comment:
Change the optional to plain properties, and check that your properties collection is not null. (Objects.requireNotNull(..)) is an easy one-liner for that.